nxdrive/gui/application.py

```diff
@@ -837,11 +841,39 @@ def _handle_macos_event(self, url: str) -> bool:
         elif cmd == "trigger-watch":
             for engine in manager._engine_definitions:
                 manager.osi.watch_folder(engine.local_folder)
+        elif cmd == "token":
+            self.api.handle_token(info["token"])
         else:
             log.warning("Unknown event URL=%r, info=%r", url, info)
             return False
         return True
 
+    def init_nxdrive_listener(self) -> None:
+        self._nxdrive_listener = QLocalServer()
+        self._nxdrive_listener.newConnection.connect(self._handle_connection)
+        self._nxdrive_listener.listen("com.nuxeo.drive.protocol")
+        self.aboutToQuit.connect(self._nxdrive_listener.close)
+
+    def _handle_connection(self) -> None:
+        con: QLocalSocket = self._nxdrive_listener.nextPendingConnection()
+        log.debug("Receiving socket connection for nxdrive protocol handling")
+        if not con or not con.waitForConnected():
+            log.debug("Unable to open server socket")
```

Comment:
Perhaps a `log.error` would be better? And maybe can we allso have a reason to help debugging: https://doc.qt.io/qt-5/qlocalsocket.html#error and/or https://doc.qt.io/qt-5/qiodevice.html#errorString?